cmd/internal/cli/build.go

```diff
@@ -297,9 +297,29 @@ func checkBuildTarget(path string) error {
 			}
 			force = true
 		}
+
+		// Remove the old image if its a SIF.
+		if !sandbox {
+			sylog.Debugf("Removing the old image: %s", path)
+			if err := os.Remove(path); err != nil {
+				return fmt.Errorf("unable to remove old image: %s", err)
+			}
+		}
 	} else if os.IsNotExist(err) && update && sandbox {
 		return fmt.Errorf("could not update sandbox %s: doesn't exist", path)
 	}
+
+	if !sandbox && !update {
+		// Create the new image now, before any building, (and umask changes).
+		sylog.Debugf("Creating output image: %s", path)
+		fp, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0755)
```

Comment:
`0755` is the perm before umask and works because it's not impacted by build umask `0002`, the quickest fix would be to just change https://github.com/sylabs/singularity/blob/master/internal/pkg/build/build.go#L78 to :

```
oldumask := syscall.Umask(0002)
defer syscall.Umask(oldumask)
```

Regarding umask for cache, I think it would also be fixed with above change

Reply:
Yup, that worked! much more simpler then I was trying to make it :smile: